main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/NativePropertyEditor/ComponentModelEditorProvider.cs

```diff
@@ -0,0 +1,136 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#if MAC
+
+using System.Linq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xamarin.PropertyEditing;
+using MonoDevelop.Core;
+
+namespace MonoDevelop.DesignerSupport
+{
+	class ComponentModelEditorProvider
+	: IEditorProvider
+	{
+		public IReadOnlyDictionary<Type, ITypeInfo> KnownTypes {
+			get;
+		} = new Dictionary<Type, ITypeInfo> ();
+
+		object [] providers;
+
+		public Task<IObjectEditor> GetObjectEditorAsync (object item)
+		{
+			if (item is ComponentModelTarget propertyPadItem) {
+				//on each editor we store current used providers
+				providers = propertyPadItem.Providers ?? Array.Empty<object> ();
+				return Task.FromResult<IObjectEditor> (new ComponentModelObjectEditor (propertyPadItem));
+			}
+			throw new ArgumentException (string.Format ("Cannot get editor for type {0} only ComponentModelObjectEditor types are allowed", item.GetType ().Name));
+		}
+
+		public Task<object> CreateObjectAsync (ITypeInfo type)
+		{
+			var realType = Type.GetType ($"{type.NameSpace}.{type.Name}, {type.Assembly.Name}");
+			if (realType == null)
+				return Task.FromResult<object> (null);
+			return Task.FromResult (Activator.CreateInstance (realType));
+		}
+
+		public Task<IReadOnlyCollection<IPropertyInfo>> GetPropertiesForTypeAsync (ITypeInfo type)
+			=> Task.Run (() => (IReadOnlyCollection<IPropertyInfo>)GetPropertiesForProviders (providers));
+
+		public static IReadOnlyList<DescriptorPropertyInfo> GetPropertiesForProviders (object[] providers)
+		{
+			var collection = new List<DescriptorPropertyInfo> ();
+
+			foreach (object propertyProvider in providers) {
+				//get the current properties for this provider
+				var currentType = propertyProvider.GetType ();
+
+				//we want all property descriptors for this propertyProvider type
+				var propertyDescriptors = System.ComponentModel.TypeDescriptor.GetProperties (currentType);
+
+				foreach (System.ComponentModel.PropertyDescriptor propertyDescriptor in propertyDescriptors) {
+					if (propertyDescriptor.IsBrowsable) {
+						collection.Add (CreatePropertyInfo (propertyDescriptor, propertyProvider));
+					}
+				}
+			}
+			return collection.AsReadOnly ();
+		}
+
+		protected static DescriptorPropertyInfo CreatePropertyInfo (System.ComponentModel.PropertyDescriptor propertyDescriptor, object PropertyProvider)
+		{
+			var valueSources = ValueSources.Local | ValueSources.Default;
+			if (propertyDescriptor.PropertyType.IsEnum) {
+				if (propertyDescriptor.PropertyType.IsDefined (typeof (FlagsAttribute), true))
```

Comment:
enums don't support inheritance

```suggestion
				if (propertyDescriptor.PropertyType.IsDefined (typeof (FlagsAttribute), false))
```

Reply:
Fixed